File: day1/Day1.kt
package day1

import java.nio.file.Files
import kotlin.io.path.Path

fun main() {
    val lines = Files.readAllLines(Path("day1/input.txt"))
    puzzle1(lines)
    puzzle2(lines)
}

private fun puzzle1(lines: List<String>) {
    var maxCalories = 0
    var current = 0
    lines.forEach {
        if (it.isBlank()) {
            if (current > maxCalories) {
                maxCalories = current
            }
            current = 0
        } else {
            current += it.toInt()
        }
    }
    println("puzzle1: $maxCalories")
}

private fun puzzle2(lines: List<String>) {
    val maxCaloriesList = mutableListOf(0, 0, 0)
    var current = 0
    lines.forEach {
        if (it.isBlank()) {
            maxCaloriesList.sort()
            if (current > maxCaloriesList.first()) {
                maxCaloriesList[0] = current
            }
            current = 0
        } else {
            current += it.toInt()
        }
    }
    println("puzzle2: ${maxCaloriesList.sum()}")
}

File: day1/day1.go
package main

import (
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {
	content, _ := os.ReadFile("./input.txt")
	lines := strings.Split(string(content), "\n")
	fmt.Println(puzzle1(lines))
	fmt.Println(puzzle2(lines))
}

func puzzle1(lines []string) int {
	var maxCalories = 0
	var currentCalories = 0
	for _, line := range lines {
		if line != "" {
			n, _ := strconv.Atoi(line)
			currentCalories += n
		} else {
			if currentCalories > maxCalories {
				maxCalories = currentCalories
			}
			currentCalories = 0
		}
	}
	return maxCalories
}

func puzzle2(lines []string) int {
	var maxCalories = []int{0, 0, 0}
	var currentCalories = 0
	for _, line := range lines {
		if line != "" {
			n, _ := strconv.Atoi(line)
			currentCalories += n
		} else {
			sort.Ints(maxCalories)
			if currentCalories > maxCalories[0] {
				maxCalories[0] = currentCalories
			}
			currentCalories = 0
		}
	}
	return maxCalories[0] + maxCalories[1] + maxCalories[2]
}
